### PR TITLE
ref(ts) Convert UserBadge to typescript

### PR DESCRIPTION
**Pull request by markstory**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/idBadge/userBadge.tsx [hunk 1]
```diff
@@ -1,48 +1,73 @@
...
+        user.email ||
+        user.username ||
+        user.ipAddress ||
+        // Because this can be used to render EventUser models, or User *interface*
+        // objects from serialized Event models. we try both ipAddress and ip_address.
+        user.ip_address)) ||
+    (member && member.name);
+
   return (
-    <StyledUserBadge {...props}>
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/idBadge/userBadge.tsx]: Do we have any components that try to style `UserBadge` via `styled`?

File: src/sentry/static/sentry/app/components/idBadge/userBadge.tsx [hunk 2]
```diff
@@ -89,16 +111,21 @@ const StyledEmail = styled('div')`
...
   return useLink ? <Link to={to} {...forwardProps} /> : <span {...forwardProps} />;
 })`
-  font-weight: ${p => (p.hideEmail ? 'inherit' : 'bold')};
+  font-weight: ${(p: NameProps) => (p.hideEmail ? 'inherit' : 'bold')};
   line-height: 1.15em;
   ${overflowEllipsis};
 `;
 
-const StyledAvatar = styled(props => <UserAvatar {...props} />)`
+const StyledAvatar = styled((props: UserAvatar['props']) => <UserAvatar {...props} />)`
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/idBadge/userBadge.tsx]: Can this just be `styled(UserAvatar)`

File: src/sentry/static/sentry/app/components/idBadge/userBadge.tsx [hunk 3]
```diff
@@ -89,16 +111,21 @@ const StyledEmail = styled('div')`
...
+  useLink: boolean;
+  hideEmail: boolean;
+} & Link['props'];
+
+const StyledName = styled(({useLink, to, ...props}: NameProps) => {
   const forwardProps = omit(props, 'hideEmail');
   return useLink ? <Link to={to} {...forwardProps} /> : <span {...forwardProps} />;
 })`
-  font-weight: ${p => (p.hideEmail ? 'inherit' : 'bold')};
+  font-weight: ${(p: NameProps) => (p.hideEmail ? 'inherit' : 'bold')};
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/idBadge/userBadge.tsx]: Can we use the generics form with `styled()`? I believe it'll be required with emotion 10's type checking. Using generics should make this redundant.

File: src/sentry/static/sentry/app/components/idBadge/userBadge.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/idBadge/userBadge.tsx]: We do. I'll add the className prop back in.

**Review by billyvg — APPROVED** [review]

**Pull request merged**